project-extender/src/providers/jira-server-provider/JiraServerProvider.ts

```diff
@@ -26,29 +42,88 @@ class JiraServerProvider implements ProviderApi {
   }
 
   async isLoggedIn(): Promise<void> {
+    const response = fetch(
+      `http://${this.requestBody.url}/rest/auth/1/session`,
+      {
+        method: "GET",
+        headers: {
+          Accept: "application/json",
+          Authorization: `Basic ${Buffer.from(
+            `${this.requestBody.username}:${this.requestBody.password}`
+          ).toString("base64")}`,
+        },
+      }
+    )
+    const GoodResponse = await response
     return new Promise((resolve, reject) => {
-      this.provider
-        ?.getCurrentUser()
-        .then(() => resolve())
-        .catch(() => reject())
+      if (GoodResponse.status === 200) resolve()
+      if (GoodResponse.status === 401) {
+        reject(new Error("Wrong Username or Password"))
+      }
+      response.catch((err) => {
+        if (err.message === "Couldn't connect to server")
+          reject(new Error("Wrong URL"))
+      })
     })
   }
 
-  async getProjects() {
-    const projectsData = await this.provider?.getProjects("1")
-    const projects = projectsData?.values.map(
-      ({ key, name }: { key: string; name: string }) => ({
-        key,
-        name,
-      })
+  async getProjects(): Promise<Project[]> {
+    const response = await fetch(
+      `http://${this.requestBody.url}/rest/api/2/project?expand=lead,description`,
+      {
+        method: "GET",
+        headers: {
+          Accept: "application/json",
+          Authorization: `Basic ${Buffer.from(
+            `${this.requestBody.username}:${this.requestBody.password}`
+          ).toString("base64")}`,
+        },
+      }
     )
+    if (response.ok) {
+      const data = await response.json()
+      const projects = data.map((project: Project) => ({
+        Key: project.key,
+        Name: project.name,
+        Type: project.projectTypeKey,
+        Lead: project.lead.displayName,
+      }))
+      return projects
+    }
+    return Promise.reject(new Error(response.statusText))
+  }
 
-    return projects
+  async logout(): Promise<void> {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const response = fetch(
+      `http://${this.requestBody.url}/rest/auth/1/session`,
+      {
+        method: "DELETE",
+        headers: {
+          Authorization: `Basic ${Buffer.from(
+            `${this.requestBody.username}:${this.requestBody.password}`
+          ).toString("base64")}`,
+        },
+      }
+    )
+    const GoodResponse = await response
+
+    return new Promise((resolve, reject) => {
+      if (GoodResponse.status === 204) {
+        resolve()
+      }
+      if (GoodResponse.status === 401)
+        reject(new Error("user not authenticated"))
+    })
```

Comment:
```suggestion
  async logout(): Promise<void> {
    return new Promise((resolve, reject) => {
      fetch(`http://${this.requestBody.url}/rest/auth/1/session`, {
        method: "DELETE",
        headers: {
          Authorization: `Basic ${Buffer.from(
            `${this.requestBody.username}:${this.requestBody.password}`
          ).toString("base64")}`,
        },
      }).then((res) => {
        if (res.status === 204) {
          resolve()
        }
        if (res.status === 401) {
          reject(new Error("user not authenticated"))
        }
      })
    })
  }
```

Fetch returns a promise in itself. But as we don't need everything returned from the jira api, we can just override the `then` clause and wrap it in a new promise.